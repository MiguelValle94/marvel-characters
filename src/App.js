import React, { Suspense } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Header from './components/Header';
import Footer from './components/Footer';
import Spinner from './components/Spinner';

import './App.scss';

const CharacterListContainer = React.lazy(() =>
  import('./components/CharacterListContainer')
);
const CharacterDetailsContainer = React.lazy(() =>
  import('./components/CharacterDetailsContainer')
);

export default function App() {
  return (
    <Router>
      <div className='app-container'>
        <header>
          <Header />
        </header>
        <main className='main'>
          <Switch>
            <Suspense fallback={<Spinner />}>
              <Route exact path='/' component={CharacterListContainer} />
              <Route path='/:id' component={CharacterDetailsContainer} />
            </Suspense>
          </Switch>
        </main>
        <footer>
          <Footer />
        </footer>
      </div>
    </Router>
  );
}
