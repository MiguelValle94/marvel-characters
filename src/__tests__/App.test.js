import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';
import { mockStore } from '../state/__mocks__/store';

const setUpShallow = (initialState) => {
  const store = mockStore(initialState);
  const component = shallow(<App store={store} />);
  return component;
};

describe('App', () => {
  it('should match snapshot', () => {
    const charactersReducer = {
      characters: [],
      isLoading: false,
      error: {}
    };

    const component = setUpShallow({ charactersReducer });
    expect(component).toMatchSnapshot();
  });
});
