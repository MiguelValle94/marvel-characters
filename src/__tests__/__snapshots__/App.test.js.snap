// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<BrowserRouter>
  <div
    className="app-container"
  >
    <header>
      <Header />
    </header>
    <main
      className="main"
    >
      <Switch>
        <Suspense
          fallback={<Spinner />}
        >
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/:id"
          />
        </Suspense>
      </Switch>
    </main>
    <footer>
      <Footer />
    </footer>
  </div>
</BrowserRouter>
`;
